components/callbacks.py

```diff
@@ -591,7 +616,10 @@ async def job_callback(_: ContextTypes.DEFAULT_TYPE) -> None:
         when=60,
     )
 
-    # We don't want this message to be processed any further
+    # We don't want this message to be processed any further, if the request didn't work out
+    # we want to log it
+    if r and r.status_code != 200:
+        r.raise_for_status()
```

Comment:
why not simply move `r.raise_for_status()` to after line 596?

Reply:
Cause then the whole job queue deletion doesn't happen

Reply:
Ahy in case the pastebin creation fails, you want to retain the current functionality? Good point. In that case, I would not see a failed pastebin request as handler callback failure though, i.e. I wouldn't let it be handled by the error handler. I would rather suggest something like


```python
if r.status != HTTPStatus.OK:
    logging.info("pastebin creation failed, I'll simply warn the user", exec_info=...)
else:
    text = ...

await message.reply_text(...)
run_once(...)
```